services/gateway/namespaces.js

```diff
@@ -13,15 +13,37 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
+const config = require('./config');
 const registerApi = require('./shared/registerRpcApi');
 
 const defaultConfig = {
 	whitelist: [],
 	aliases: {},
 };
 
-module.exports = {
-	'/rpc': registerApi('http-version1', { ...defaultConfig }),
-	'/rpc-v1': registerApi('http-version1', { ...defaultConfig }),
-	'/rpc-test': registerApi('http-test', { ...defaultConfig }),
-};
+const enableApiWS = config.api.ws.split(',');
+const exportedApi = {};
+
+enableApiWS.forEach(apiName => {
+	if (apiName === 'rpc') {
+		exportedApi['/rpc'] = registerApi('http-version1', { ...defaultConfig });
+	}
+	if (apiName === 'rpc-v1') {
+		exportedApi['/rpc-v1'] = registerApi('http-version1', { ...defaultConfig });
+	}
+	if (apiName === 'rpc-test') {
+		exportedApi['/rpc-test'] = registerApi('http-version1', { ...defaultConfig });
+	}
+	if (apiName === 'blockchain') {
+		exportedApi['/blockchain'] = {
+			events: {
+				call: {
+					mappingPolicy: 'restrict',
+					aliases: {},
+				},
+			},
+		};
+	}
+});
```

Comment:
This spot deserves some functional approach here:

```javascript
module.exports = filterApis(config.api.ws, {
	'/rpc': registerApi('http-version1', { ...defaultConfig }),
	'/rpc-v1': registerApi('http-version1', { ...defaultConfig }),
	'/rpc-test': registerApi('http-test', { ...defaultConfig }),
});
```

And the write a function that picks proper apis from that list.

To prevent `registerApi` from running at start, wrap in an anonymous function:
```
() => registerApi('http-version1', { ...defaultConfig })
```